src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3764,24 +3764,24 @@ private void AssertQueryDoesNotContainFieldsThatAreNotIndexed(QueryMetadata meta
                         continue;
 #endif
 
-                    AssertKnownField(f);
+                    AssertKnownField(f, metadata.IndexName);
                 }
             }
         }
 
-        private void AssertKnownField(string f)
+        private void AssertKnownField(string f, string indexName)
         {
             // the catch all field name means that we have dynamic fields names
 
             if (Definition.HasDynamicFields || IndexPersistence.ContainsField(f))
                 return;
 
-            ThrowInvalidField(f);
+            ThrowInvalidField(f, indexName);
         }
 
-        private static void ThrowInvalidField(string f)
+        private static void ThrowInvalidField(string f, string indexName)
         {
-            throw new ArgumentException($"The field '{f}' is not indexed, cannot query/sort on fields that are not indexed");
+            throw new ArgumentException($"The field '{f}' is not indexed for index {indexName}, cannot query/sort on fields that are not indexed");
```

Comment:
quote the index name and check failing tests

Reply:
done